### ISSUE TITLE
Payment Flow

### ISSUE THREAD

**Issue by achille005** [achille005/JobApp#10]
- Create Job Profile > Payment screen  ====> is WRONG!
Here is the correct flow:
- Any button is clicked on Recruiter Dashboard 
> Verify Recruiter has a current paid plan:
           Yes > allow to use all feature
            No > take to payment screen:
4 plan types:
Each plan includes unlimited editing, resumes download(i), Contact Job Seeker

             1 month @ $250.00 --> Up to 10 Job Profiles setup

              3 months @ $725.00 --> Up to 20 Job Profiles setup

              6 months @ $1,450.00  --> Up to 30 Job Profiles setup

              12 months @ $2,875.00  --> Up to 50 Job Profiles setup

> A plan is selected > Authorize payment > payment Successful > Take user to dashboard > User can use any feature for (1, 3, 6 or 12 months) depending on plan purchased

(i) If resume is not submitted "download/forward resume" button is grayed out but you may contact the Job Seeker

**Comment by Astrea123** [comment]
can you create a flow chart explain the flow of payment, because 3 payment options was included in previous flow

**Comment by Astrea123** [comment]
When you say
 1 month @ $250.00 --> Up to 10 Job Profiles setup

It means user can post upto 10 job for 1 month validation  and user will be pay only one time  ?
And payment flow will be apply when click on new job post ?

**Comment by achille005** [comment]
Exactely! if the user attempt to post 11th job in that month he paid 250.0 for the app will ask him if he wants to upgrade? After 1 month though his subscription becomes expired.

**Comment by Astrea123** [comment]
I mean if user has post one job already with the registration process. after that user try to use second job then he will see the payment screen ?

**Comment by achille005** [comment]
No. User can post up to 10 jobs for free! 11th is paid.  but even if user post 1 jobs and click any of those 3 buttons on dashboard check if he has paid account? if not take him  to payment packages screen

**Comment by Astrea123** [comment]
ok ,In that case  if user has not paid account then we should not show current plan like 
Current Plan: 1 month @ $250.00 --> Up to 10 Job Profiles setup

**Comment by achille005** [comment]
True!

**Comment by Astrea123** [comment]
One more doubt ,User will be pay only one time for all the job he will not pay for each job right ?In that case which job info we will show on the Payment screen page

**Comment by Astrea123** [comment]
payment plan info will save in the contact table

**Comment by achille005** [comment]
- One more doubt ,User will be pay only one time for all the job he will not pay for each job right?
Yes
- In that case which job info we will show on the Payment screen page?
Show the plan (all jobs go under this plan). On Summary page says:
You are posting to: "Tech Jobs: IT & Software" Channel 
                                              for
                                1 month @ $250.00 --> Up to 10 Job Profiles setup

- payment plan info will save in the contact table?
I think so - it is for the R - and there are 4 plan types

**Comment by achille005** [comment]
How is this going?

**Comment by Astrea123** [comment]
I have done this .Today I will deploy the package .

**Comment by achille005** [comment]
On this screen where the there Bulk price correct following:
- 1 Bulk Pricing 
It is:  1 month @ $250.00 --> Up to 10 Job Profiles setup and NOT
1 month @ $250.00 each --> Up to 10 Job Profiles setup
(Notice the "each" in the 4 lines )
- 2 - Replace the arrow --> by ==> 
- 3 - Change the page Title to "Payment"
- 4 - Continue button
Make sure Continue is pressed before the page navigate to next Screen.
- 5 - On next Screen, say the pricing information should be:
Price: 1 month @ $250.00  ==> Up to 10 Job Profiles setup
(Include: Unlimited editing, Resumes viewing/downloading/Forwarding, Email candidates) 
You are posting to: "**Tech Jobs: IT & Software**" Category in **USA, Atlanta, GA** (Change)
Where "Change" is a link that takes the user back to "Choose a Category" screen 

- 6- Please provide the test credit card data

**Comment by Astrea123** [comment]
I have a doubt .....
    We are storing country,state,city on Job Profile page  .There is no country,state , city on recruiter 
     information page .So from where we will show Category in **USA, Atlanta, GA** on payment page

**Comment by achille005** [comment]
I am WRONG here! Sorry for the confusion.
Just show this:
**Price**: 1 month @ $250.00 ==> Up to 10 Job Profiles setup
_(Include: Unlimited editing, Resumes viewing/downloading/Forwarding, Email candidates)_

**Comment by achille005** [comment]
- IGNORE all the rules before this one( welcome to water fall-- It will be rewarded!):
- It will be nice if we can wrap up this app by the end of this month -  I have found this link( https://stripe.com/docs/subscriptions/trials) for subscription with 3 months free trial period. To keep it simple we want to set the same monthly reoccurring price for everybody $95.00 till cancel for everybody

Sample Payment Screen that comes after Contact this Job Seeker is clicked
================================================
![image](https://user-images.githubusercontent.com/6147947/28219284-7235bc34-6889-11e7-9b3f-d0221324da4d.png)

**Comment by Astrea123** [comment]
We have doubt in this that you want free trial of stripe for 3 months for yourself or you want recruiter to have free trial of 3 month, at the end of which user has to make payment.

**Comment by achille005** [comment]
Recruiter will have 3 months free trial. not me with stripe

**Comment by Astrea123** [comment]
Do you want to change existing GUI of payment page also?

**Comment by Astrea123** [comment]
I have checked https://stripe.com/docs/subscriptions/trials.
I got your point we are exploring this topic. but for doing this we need to make changes in the flow and I have following doubt

1) Now when we provide trial account then we cannot restrict user after login to download,forward  resume,contact jobseeker
So when we include payment window

2) After trial end How to allow to make payment from customer

3)If Recruiter does not want to extend plan from free to monthly then what should we do
 
4) If User does not want to enter info of card then should we allow to login user

**Comment by achille005** [comment]
- Do you want to change existing GUI of >
You don't have to change the input Fields. Just add the text About payment, type of CC used i hv added

- Now when we provide trial account then we cannot restrict user after login to download,forward..

Sure, during the trial everything should be free.

- After trial end How to allow to make payment from customer

The Stripe API should know how to handle that. you just to the setting with trial_end var

- If Recruiter does not want to extend plan from free to monthly then what should we do

They cannot contact JS, download and foward. They can still receive cron jobs notification

- If User does not want to enter info of card then should we allow to login user

The user should be already logged in to be able to get to payment flow. recall the flow? after login > Dashboard > Recruiter touch contact this Job Seeker/download/forward resume > Payment flow
Note: if you have never done anything like that please get help. If you can find help please let us know.

**Comment by Astrea123** [comment]
**With this point "After trial end How to allow to make payment from customer" we mean to say that**,

 3 days prior to expiration of recruiters time period, stripe will send him mail reminding that his subscription is about to expire
Now there are 2 ways to handle the payment 
1)Then After 3 month payment will automatically cut from the account of user since we have already card info of user
2)Show payment card to the user again and after filling that form payment will be done

And there is one more possible condition more 
1) if user does not want to continue after trial account ,he/she wants to cancel the subscription then how will be show in the app

**Comment by achille005** [comment]
currently we have the statuses:
-Daily
-weekly
-OFF
OFF should cancel everything because there will be no alertst sent to inbox to click on if status is OFF.
you can do that from dashboard today



Admin control this:
-Active
-Deactivated

**Comment by Astrea123** [comment]
This is the flowchart we have made for payment functionality, please tell if it is fine, or you want something else.

https://drive.google.com/file/d/0B5yTvhVBvdrqOTZSUEhvU1luZU0/view

**Comment by achille005** [comment]
that looks right. To what i know you don't manually implement that but you setup don't property or plug in those code snippet from stripe and they take care of it.
